graph/push.go

```diff
@@ -428,8 +435,9 @@ func (s *TagStore) CmdPush(job *engine.Job) engine.Status {
 			return engine.StatusOK
 		}
 
-		// error out, no fallback to V1
-		return job.Errorf("Error pushing to registry: %s", err)
+		if err != ErrV2RegistryUnavailable {
+			return job.Errorf("Error pushing to registry: %s", err)
+		}
 	}
 
 	if err != nil {
```

Comment:
I wish I had a better answer for you than because it was already there.  This err check is from the previous call to graph.Get() on line 422.  In a previous iteration there was likely a reason for this separation.  We chose not to include cleanup/refactor of v1 code as part of inserting the v2 code though to keep the diffs smaller.